### ISSUE TITLE
MODE_WORLD_READABLE no longer supported

### ISSUE THREAD

**Issue by snodnipper** [wiglenet/wigle-wifi-wardriving#100]
On Android API 25 I get the message as shown below when trying to upload.
Looks like [runtime permissions](https://developer.android.com/training/permissions/requesting.html) could be added.

![screenshot_20170222-160239](https://cloud.githubusercontent.com/assets/576719/23220200/12a4661a-f919-11e6-9bb5-ce1c76e90ec2.png)

**Comment by bobzilladev** [comment]
Things work ok on other 7.1.1 devices. I would guess at first run you said 'yes' to the pop-up asking for permission to write to the virtual sd card, but later revoked that permission? So now it's configured to write to the sd card but can't. Solution would be to either go in the settings app and give wiglewifi the permission, or re-install and say 'no' to the pop-up question. We don't have a way to configure that after that pop-up (maybe we should add that, or always fall-back to non-sdcard app storage).

**Comment by snodnipper** [comment]
> 'yes' to the pop-up asking for permission to write to the virtual sd card, but later revoked that permission?

I think that I denied the permission first...and possibly selected "Don't ask again."

I actually checked out the code and managed to patch it (v.quickly)...first manually with a snackbar...but then I gave [EasyPermissions](https://github.com/googlesamples/easypermissions) a go, which seemed tidier.  It also gives a generic message if folks select "Don't ask again", which can take them to the app settings.  I can possibly take another look this weekend if helpful.
